### PR TITLE
Add procedures on how to import ansible playbooks

### PR DESCRIPTION
**Pull request by Levi-Leah**
Add procedures on how to import ansible playbooks with API

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [ ] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+[id="importing-all-available-ansible-playbooks_{context}"]
+= Importing All Available Ansible Playbooks
+
+You can import all the available Ansible playbooks on your host.
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:5]:
```suggestion
You can import all the available Ansible playbooks from collections installed on the Capsule.
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+[id="importing-an-ansible-playbook-by-name_{context}"]
+= Importing an Ansible Playbook by Name
+
+You can import an Ansible playbook on your host by name.
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:5]:
```suggestion
You can import an Ansible playbook on your host by name from collections installed on the Capsule.
```

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+[id="importing-all-available-ansible-playbooks_{context}"]
+= Importing All Available Ansible Playbooks
+
+You can import all the available Ansible playbooks from collections installed on the Capsule.
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:5]:
```suggestion
You can import all the available Ansible playbooks from collections installed on the {SmartProxy.
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+
+[id="importing-an-ansible-playbook-by-name_{context}"]
+= Importing an Ansible Playbook by Name
+
+You can import an Ansible playbook on your host by name from collections installed on the Capsule.
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:5]:
```suggestion
You can import an Ansible playbook on your host by name from collections installed on the {SmartProxy}.
```

**Review by ares — APPROVED** [review]
LGTM (content wise, I can't weigh in on the asciidoc), the only thing is regarding the URL in the example (not sure what it renders to)

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+[id="importing-all-available-ansible-playbooks_{context}"]
+= Importing All Available Ansible Playbooks
+
+You can import all the available Ansible playbooks from collections installed on the {SmartProxy}.
+
+.Procedure
+. On the target host, import the Ansible playbooks:
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X PUT 'Content-Type: application/json' https://_{smartproxy-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/sync?proxy_id=_proxy_id_number_ | json_pp
```
Comment (ares) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:11]: I'm not entirely sure what smartproxy-example-com and smartproxy_port resolves to, but it's the Foreman's API and port right? Similarly in the other guide.

Comment (ekohl) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:11]:
```console
$ rg 'smartproxy-example-com|smartproxy_port' guides/common/attributes*
guides/common/attributes-satellite.adoc
87::smartproxy_port: 9090
88::smartproxy-example-com: capsule.example.com

guides/common/attributes-orcharhino.adoc
45::smartproxy-example-com: orcharhino-proxy.example.com

guides/common/attributes-katello.adoc
6::smartproxy_port: 9090

guides/common/attributes-base.adoc
98::smartproxy_port: 8443
99::smartproxy-example-com: smartproxy.example.com
```

**Comment by ekohl** [comment]
> the only thing is regarding the URL in the example (not sure what it renders to)

See the rendered version (and diffs on rendered content) in https://github.com/theforeman/foreman-documentation/pull/1746#issuecomment-1297415019.

I do wonder if you intended to add it to the AppCentric guide and not some other guide.

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc (hunk 5 above)
Comment (ekohl) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:11]: It's very uncommon to connect to a Smart Proxy, especially when you also pass a proxy ID. Didn't you mean `{foreman-example-com}` (and leave out the port)?

Comment (Lennonka) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:11]: Yes, the API call should be pointed at a SatServer URL.

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 6]
```diff
@@ -0,0 +1,13 @@
+
+[id="importing-all-available-ansible-playbooks_{context}"]
+= Importing All Available Ansible Playbooks
+
+You can import all the available Ansible playbooks from collections installed on the {SmartProxy}.
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:5]:
```suggestion
You can import all the available Ansible playbooks to {Project} from collections installed on {SmartProxy}.
You can then use the playbooks to invoke jobs using the _Ansible - Run playbook_ job template.
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 7]
```diff
@@ -0,0 +1,21 @@
+
+[id="importing-an-ansible-playbook-by-name_{context}"]
+= Importing an Ansible Playbook by Name
+
+You can import an Ansible playbook from collections installed on the {SmartProxy}  by name.
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:5]:
```suggestion
You can import an Ansible playbook to {Project} from collections installed on {SmartProxy} by name.
You can then use the playbook to invoke jobs using the _Ansible - Run playbook_ job template.
```

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+.Procedure
+. On the target host, import the Ansible playbooks:
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:8]:
```suggestion
. Import the Ansible playbooks using the following API request:
```
The API call does not have to be made from a host and it does not relate to a particular host. (API calls can be technicaly made from a local machine or anywhere that can access the SatServer over HTTPS.)

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+.Procedure
+. On the target host, fetch the available Ansible playbooks.
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:8]:
```suggestion
. Fetch the available Ansible playbooks using the following API request:
```

**Review by Lennonka — CHANGES_REQUESTED** [review]
Hi Levi!
The playbooks relate to job templates, which are used to execute remote jobs on hosts.
The user imports them so that they can later use them to run remote jobs using the _Ansible - Run playbook_ job template, which already exists in Project.
I would see the procedures after [Creating Job Templates](https://docs.theforeman.org/nightly/Managing_Hosts/index-satellite.html#creating-a-job-template_managing-hosts).
Also see my other comments. I hope this helps to clarify things.

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 10]
```diff
@@ -0,0 +1,22 @@
+include::attributes-base.adoc[]
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:1]:
```suggestion
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc (hunk 7 above)
Comment (Lennonka) [reply guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc]: Do you have a reason not to include the second line in the suggestion?

Comment (Levi-Leah) [reply guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc]: > Do you have a reason not to include the second line in the suggestion?
@Lennonka I think it's common knowledge in this case. The user importing Ansible playbooks would likely know how to use them

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 11]
```diff
@@ -1,21 +1,22 @@
...
```
Comment (maximiliankolb) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:1]:
```suggestion
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 12]
```diff
@@ -1,21 +1,22 @@
...
-curl -X GET 'Content-Type: application/json' https://_{smartproxy-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/fetch?proxy_id=_proxy_id_number_ | json_pp
+curl -X GET 'Content-Type: application/json' https://_{foreman-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/fetch?proxy_id=_proxy_id_number_ | json_pp
 ----
 . Select the Ansible playbook you want to import and note its name.
 . Import the Ansible playbook using its name:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_playbook_name_"] }' https://_{smartproxy-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/sync?proxy_id=_proxy_id_number_ | json_pp
+curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_playbook_name_"] }' https://_{foreman-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/sync?proxy_id=_proxy_id_number_ | json_pp
```
Comment (maximiliankolb) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc]:
```suggestion
curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_My_Playbook_Name_"] }' https://_{foreman-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_smart-proxy-context_ID_ | json_pp
```

With Hammer CLI, we mostly prefix `My_` to all values. I think this makes sense here as well.

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+.Prerequisite
+* Ansible plug-in in {Project} is enabled
+
+.Procedure
+. Fetch the available Ansible playbooks using the following API request:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X GET 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/fetch?proxy_id=_My_smart-proxy-context_ID_ | json_pp
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:15]:
```suggestion
curl -X GET 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/fetch?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+----
+. Select the Ansible playbook you want to import and note its name.
+. Import the Ansible playbook using its name:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_My_Playbook_Name_"] }' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_smart-proxy-context_ID_ | json_pp
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:22]:
```suggestion
curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_My_Playbook_Name_"] }' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+.Prerequisite
+* Ansible plug-in in {Project} is enabled
+
+.Procedure
+. Import the Ansible playbooks using the following API request:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_smart-proxy-context_ID_ | json_pp
```
Comment (Lennonka) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:15]:
```suggestion
curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```

**Review by Lennonka — CHANGES_REQUESTED** [review]
Just one more tweak -- smart-proxy-context is supposed to be an attribute.

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+[id="importing-all-available-ansible-playbooks_{context}"]
+= Importing All Available Ansible Playbooks
+
+You can import all the available Ansible playbooks to {Project} from collections installed on {SmartProxy}.
+
+.Prerequisite
+* Ansible plug-in in {Project} is enabled
+
+.Procedure
+. Import the Ansible playbooks using the following API request:
```
Comment (ekohl) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:11]: I was wondering: isn't this available in [hammer-cli-foreman-ansible](https://github.com/theforeman/hammer-cli-foreman-ansible). Looking at the source I only see one hit on `sync` (https://github.com/theforeman/hammer-cli-foreman-ansible/search?q=sync) and that's in https://github.com/theforeman/hammer-cli-foreman-ansible/blob/a2a884f1799e393f8a4ea340bc60c48dd8e2acc1/lib/hammer_cli_foreman_ansible/ansible_roles.rb#L28-L35

Given the filename and class name I get the impression it's only for roles, not playbooks. If we don't have this functionality, I think that deserves an issue to add it. Providing a CLI command is a better experience for most users than calling the API.

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```
Comment (ekohl) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:15]: Where does `json_pp` come from? We don't have that anywhere in our current docs.

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc (hunk 16 above)
Comment (Lennonka) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:11]: AFAIK, it's only available via API, not other UIs yet.

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc (hunk 17 above)
Comment (Levi-Leah) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:15]: > Where does `json_pp` come from? We don't have that anywhere in our current docs.

from the demo https://www.youtube.com/watch?v=ykQFM7aFczY&t=1788s

Comment (ekohl) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:14]: I think it's a developer short hand for JSON Pretty Print. Most people won't have that (on a related note, you can get the same by piping it to [jq](https://stedolan.github.io/jq/)). I'd leave it out

**Comment by Lennonka** [comment]
Just FYI, it should go to 3.4 as well, I'm adding a tick for it.

**Comment by Levi-Leah** [comment]
> Just FYI, it should go to 3.4 as well, I'm adding a tick for it.

Thanks!

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:14]:
```suggestion
curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+
+.Prerequisite
+* Ansible plug-in in {Project} is enabled
+
+.Procedure
+. Fetch the available Ansible playbooks using the following API request:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X GET 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/fetch?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:14]:
```suggestion
curl -X GET 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/fetch?proxy_id=_My_{smart-proxy-context}_ID_
```

File: guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+----
+. Select the Ansible playbook you want to import and note its name.
+. Import the Ansible playbook using its name:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_My_Playbook_Name_"] }' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```
Comment (Levi-Leah) [inline guides/common/modules/proc_importing-an-ansible-playbook-by-name.adoc:21]:
```suggestion
curl -X PUT 'Content-Type: application/json' -d '{ "playbook_names": ["_My_Playbook_Name_"] }' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_
```

File: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc (hunk 16 above)
Comment (ekohl) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:10]: Bonus points, but in code I like to add comments for that. Something like
```suggestion
// Not available via Hammer/UI: <link-to-issue>
. Import the Ansible playbooks using the following API request:
```

Then in the future if you're reading it you know why it's done via the API and have an easy way to check if it's been resolved yet.

Comment (Levi-Leah) [reply guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc:10]: > Then in the future if you're reading it you know why it's done via the API and have an easy way to check if it's been resolved yet.

Yeah, good idea

**Review by Lennonka — APPROVED** [review]
Happy with it. Merging once the checks pass.

**Comment by Lennonka** [comment]
Cherry-picked:
- 398731ee9..69458fb7e  3.4 -> 3.4
- 4274f6eec..e4fa38a95  3.3 -> 3.3

**Pull request merged**